### PR TITLE
docs: clarified reference to project parameter

### PR DESCRIPTION
**Pull request by craiglabenz**
Region tags `firestore_setup_client_create` and `firestore_setup_client_create_with_project_id` are being consolidated. The latter was never a thing in the Python samples, but here we keep what Python samples that do exist in line with the target state for all 7 languages (1 snippet that explains the optional parameter).

### CODE REVIEW

File: firestore/cloud-async-client/snippets.py [hunk 1]
```diff
@@ -20,8 +20,9 @@ async def quickstart_new_instance():
     # [START firestore_setup_client_create_async]
     from google.cloud import firestore
 
-    # Project ID is determined by the GCLOUD_PROJECT environment variable
-    db = firestore.AsyncClient()
+    # The `project` parameter is optional and if not supplied will
+    # fall back to your gcloud project configuration.
```
Comment (crwilcox) [inline firestore/cloud-async-client/snippets.py:24]: I may be too concerned, but the way we infer projects is broader than gcloud. https://github.com/googleapis/python-cloud-core/blob/master/google/cloud/client.py#L237

```
    # If you don't specify credentials when constructing the client, the
    # client library will look for credentials in the environment.
```

Is how we describe it in auth, https://cloud.google.com/docs/authentication/getting-started#verifying_authentication. Even that isn't very specific to what is happening though, maybe intentionally.

Comment (craiglabenz) [reply firestore/cloud-async-client/snippets.py:24]: I share this concern - went back and forth on it with @kolea2 last week on the phrasing. Maybe the three of us can put our heads together on one succinct phrasing to rule them all.

Comment (kolea2) [reply firestore/cloud-async-client/snippets.py:24]: I think you still need to update this comment to match the other?

Comment (craiglabenz) [reply firestore/cloud-async-client/snippets.py:24]: Oh my goodness - this Python a/sync split will be the end of me one day.

**Review by leahecole — APPROVED** [review]

**Comment by craiglabenz** [comment]
Like Sisyphus, I appear doomed to merge master into this branch for the rest of time.

**Comment by leahecole** [comment]
> Like Sisyphus, I appear doomed to merge master into this branch for the rest of time.

Gotta get two approvals in this repo! @crwilcox or @kolea2 should be your number 2

**Review by kolea2 — APPROVED** [review]
lgtm!

**Comment by kolea2** [comment]
@googlebot

**Pull request merged**